lib/src/builder.rs

```diff
@@ -75,9 +75,50 @@ pub fn calculate_batch_blocks_final_header(input: &GuestBatchInput) -> Vec<Block
                 .expect("execute single batched block"),
         );
     }
+    assert!(
+        verify_final_batch_blocks(input, &final_blocks).is_ok(),
+        "Invalid batch blocks"
+    );
     final_blocks
 }
 
+// to check the linkages between the blocks
+// 1. connect parent hash & state root
+// 2. block number should be in sequence
+fn verify_final_batch_blocks(input: &GuestBatchInput, final_blocks: &[Block]) -> Result<()> {
```

Comment:
The returned value is always `Ok(())`.

```suggestion
fn verify_final_batch_blocks(input: &GuestBatchInput, final_blocks: &[Block]) {
```